### PR TITLE
Improve the functionality of qmk.questions

### PR DESCRIPTION
**Pull request by skullydazed**
This adds new functionality to the `qmk.questions` module:

* Adds a validate argument to `question()` so that developers can validate answers
* Adds a new `choice()` function for selecting options from a list

## Types of Changes

- [x] Core
- [x] New feature
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/hello.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 This is an example QMK CLI script.
 """
 from milc import cli
+from qmk.questions import question
```
Comment (Erovia) [inline lib/python/qmk/cli/hello.py]: I guess you wanted to do something with this.

File: lib/python/qmk/questions.py [hunk 2]
```diff
@@ -57,17 +57,23 @@ def yesno(prompt, *args, default=None, **kwargs):
...
         default
             The value to return when the user doesn't enter any value. Use None to prompt until they enter a value.
 
         answer_type
             Specify a type function for the answer. Will re-prompt the user if the function raises any errors. Common choices here include int, float, and decimal.Decimal.
+
+        validate
+            This is an optional function that can be used to validate the answer. It should return True or False and have the following signature:
+
+                def function_name(answer, *args, **kwargs):
```
Comment (Erovia) [inline lib/python/qmk/questions.py]: Maybe we could add something like this:
```
For simple scenarios a lambda function works, too:

    lambda answer: True if answer <condition> else False
```

File: lib/python/qmk/questions.py [hunk 3]
```diff
@@ -76,17 +82,21 @@ def question(prompt, *args, default=None, confirm=False, answer_type=str, **kwar
         prompt = '%s [%s] ' % (prompt, default)
```
Comment (Erovia) [inline lib/python/qmk/questions.py]: The output this way feels more natural to me.

```suggestion
        prompt = '%s [%s]: ' % (prompt, default)
    else:
        prompt = '%s: ' % prompt
```

File: lib/python/qmk/questions.py [hunk 4]
```diff
@@ -57,17 +57,23 @@ def yesno(prompt, *args, default=None, **kwargs):
...
+def question(prompt, *args, default=None, confirm=False, answer_type=str, validate=None, **kwargs):
     """Prompt the user to answer a question with a free-form input.
 
+    Arguments:
         prompt
-            The prompt to present to the user. Can include ANSI and format strings like milc's `cli.print()`.
+            The prompt to present to the user. Can include ANSI and format strings like milc's `cli.echo()`.
 
         default
             The value to return when the user doesn't enter any value. Use None to prompt until they enter a value.
```
Comment (Erovia) [inline lib/python/qmk/questions.py]:
```suggestion
        
        confirm
            Present the user with a confirmation dialog before accepting their answer.

```

File: lib/python/qmk/cli/hello.py (hunk 1 above)
Comment (skullydazed) [reply lib/python/qmk/cli/hello.py]: Good catch, yeah. I was using that command to test.

File: lib/python/qmk/questions.py (hunk 2 above)
Comment (skullydazed) [reply lib/python/qmk/questions.py]: I'd prefer not promoting the use of lambdas. I find that often those sorts of simple checks are useful in multiple places.

File: lib/python/qmk/questions.py (hunk 3 above)
Comment (skullydazed) [reply lib/python/qmk/questions.py]: I'm leaving control of the prompt up to the programmer. They might want to use `> ` or  `- ` (or some unicode character I don't know about,) and I decided the complexity of adding another argument to control that part of the prompt was not worth the benefit. 

I may change my mind later if we have to spend a lot of time correcting this in submissions. :)

**Comment by skullydazed** [comment]
Unfinished travis job was to compile all the keymaps, which is irrelevant to python code.

**Pull request merged**